core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -387,4 +388,163 @@ public void dataFilesCleanup() throws IOException {
     Assert.assertTrue("FILE_A should be deleted", deletedFiles.contains(FILE_A.path().toString()));
     Assert.assertTrue("FILE_B should be deleted", deletedFiles.contains(FILE_B.path().toString()));
   }
+
+  /**
+   * Test on table below, and expiring the staged commit `B` using `expireOlderThan` API.
+   * Table: A - C
+   *          ` B (staged)
+   */
+  @Test
+  public void testWithExpiringDanglingStageCommit() {
+    // `A` commit
+    table.newAppend()
+        .appendFile(FILE_A)
+        .commit();
+
+    // `B` staged commit
+    table.newAppend()
+        .appendFile(FILE_B)
+        .stageOnly()
+        .commit();
+
+    TableMetadata base = readMetadata();
+    Snapshot snapshotB = base.snapshots().get(1);
+
+    // `C` commit
+    table.newAppend()
+        .appendFile(FILE_C)
+        .commit();
+
+    List<String> deletedFiles = new ArrayList<>();
+
+    // Expire all commits including dangling staged snapshot.
+    table.expireSnapshots()
+        .deleteWith(deletedFiles::add)
+        .expireOlderThan(snapshotB.timestampMillis() + 1)
+        .commit();
+
+    // Files should be deleted of dangling staged snapshot
+    snapshotB.addedFiles().forEach(item -> {
+      Assert.assertTrue(deletedFiles.contains(item.path().toString()));
+    });
+  }
+
+  /**
+   * Expire cherry-pick the commit as shown below, when `B` is in table's current state
+   *  Table:
+   *  A - B
+   *   `- C (B) <--this commit
```

Comment:
The label "this commit" makes sense in context for `RemoveSnapshots` because it is iterating through snapshots/commits. It makes less sense here. Instead, I would use "current" and note which commits are expiring (e.g., "expiring A and B").

Reply:
I think it would also be easier to understand if this explicitly labeled the source with `(source=B)` instead of just `(B)`.